src/astro/utils/database.py

```diff
@@ -0,0 +1,82 @@
+from typing import Union
+
+from airflow.hooks.base import BaseHook
+from airflow.providers.sqlite.hooks.sqlite import SqliteHook
+from sqlalchemy import create_engine, text
+from sqlalchemy.engine import Engine
+from sqlalchemy.engine.result import ResultProxy
+
+from astro.constants import Database
+from astro.utils.dependencies import BigQueryHook, PostgresHook, SnowflakeHook
+
+
+def get_database_name(interface: Union[Engine, BaseHook, SqliteHook]) -> Database:
+    """
+    Given a hook or a SQL engine, identify the database name.
+
+    :param interface: interface to the database
+    :type interface: SQLAlchemy engine or Airflow Hook (BigQueryHook, PostgresHook, SnowflakeHook, SqliteHook)
+    :return: the database this interface relates to (e.g. Database.SQLITE)
+    :rtype: astro.constants.Database enum item
+    """
+    if isinstance(interface, BaseHook):
+        hook_to_database = {
+            BigQueryHook: Database.BIGQUERY,
+            PostgresHook: Database.POSTGRES,
+            SnowflakeHook: Database.SNOWFLAKE,
+            SqliteHook: Database.SQLITE,
+        }
+        try:
+            database_name = hook_to_database[type(interface)]
+        except KeyError:
+            raise ValueError(f"Unsupported database {type(interface)}")
+    else:  # SqlAlchemy engine
+        database_name = getattr(Database, interface.name.upper())
+    return database_name
+
+
+def get_sqlalchemy_engine(hook: Union[BaseHook, SqliteHook]) -> Engine:
+    """
+    Given a hook, return a SQLAlchemy engine for the target database.
+
+    :param hook: Airflow Hook used to access a SQL-like database
+    :type hook: (BigQueryHook, PostgresHook, SnowflakeHook, SqliteHook)
+    :return: SQLAlchemy engine
+    :rtype: sqlalchemy.Engine
+    """
+    database = get_database_name(hook)
+    engine = None
+    if database == Database.SQLITE:
+        uri = hook.get_uri()
+        if "////" not in uri:
+            uri = hook.get_uri().replace("///", "////")
```

Comment:
we are not supporting windows?
https://docs.sqlalchemy.org/en/14/core/engines.html#sqlite